transifex/pull.py

```diff
@@ -6,238 +6,44 @@
-            subprocess.run(['django-admin.py', 'compilemessages'], check=True)
-        except subprocess.CalledProcessError:
-            self.compilemessages_failed = True
-
-    def is_changed(self):
-        """Determine whether any changes were made."""
-        completed_process = subprocess.run(['git', 'status', '--porcelain'], stdout=subprocess.PIPE, check=True)
-        return bool(completed_process.stdout)
-
-    def commit(self):
-        """Commit changes.
-
-        Adds any untracked files, in case new translations are added.
-        """
-        subprocess.run(['git', 'add', '-A'], check=True)
-        try:
-            subprocess.run(['git', 'commit', '-m', self.message], check=True)
-        except subprocess.CalledProcessError:
-            subprocess.run(
-                [
-                    'git',
-                    '-c', 'user.name="{}"'.format(os.environ['GIT_USER_NAME']),
-                    '-c', 'user.email={}'.format(os.environ['GIT_USER_EMAIL']),
-                    'commit', '-m', self.message
-                ],
-                check=True
-            )
 
-    def push(self):
-        """Push branch to the remote."""
-        subprocess.run(['git', 'push', '-u', 'origin', self.branch_name], check=True)
+from utils import logger, repo_context
 
-    def pr(self):
-        """Create a new PR on GitHub."""
-        return self.github_repo.create_pull(
-            self.message,
-            'This PR was created by a script.',
-            'master',
-            self.branch_name
-        )
 
-    def cleanup(self, pr):
-        """Delete the local clone of the repo.
-
-        If applicable, also deletes the merged branch from GitHub.
-        """
-        if pr and pr.is_merged():
-            logger.info('Deleting merged branch %s:%s.', self.name, self.branch_name)
-            # Delete branch from remote. See https://developer.github.com/v3/git/refs/#get-a-reference.
-            ref = 'heads/{branch}'.format(branch=self.branch_name)
-            self.github_repo.get_git_ref(ref).delete()
-
-        # Delete cloned repo.
-        subprocess.run(['rm', '-rf', self.name], check=True)
-
-
-@contextmanager
-def cd(repo):
-    """Utility for changing into and out of a repo."""
-    initial_directory = os.getcwd()
-    os.chdir(repo.name)
-
-    # Exception handler ensures that we always change back to the
-    # initial directory, regardless of how control is returned
-    # (e.g., an exception is raised while changed into the new directory).
-    try:
-        yield
-    finally:
-        os.chdir(initial_directory)
-
-
-def pull(repo):
+def pull(clone_url, skip_compilemessages=False):
     """Pulls translations for the given repo.
 
     If applicable, commits them, pushes them to GitHub, opens a PR, waits for
     status checks to pass, then merges the PR and deletes the branch.
     """
-    pr = None
-    logger.info('Pulling translations for [%s].', repo.name)
+    with repo_context(clone_url) as repo:
```

Comment:
No, I also moved lots of code around, changed the sleep time for merge retries, removed the ability to run the script without an explicit clone_url, among other things.